### PR TITLE
Added the Download filter and download badge

### PR DESCRIPTION
**Pull request by abhijeetChawla**
Added the download filter to the library and The download badge to the to the mangaCard
localhost_3000_library

### CODE REVIEW

File: src/components/library/LibraryMangaGrid.tsx [hunk 1]
```diff
@@ -23,15 +23,41 @@ function unreadFilter(unread: NullAndUndefined<boolean>, { unreadCount }: IManga
...
+//     switch (downloaded) {
+//         case true:
+//             return !!downloaded && downloadCount >= 1;
+//         case false:
+//             return downloadCount === 0;
+//         default:
+//             return true;
+//     }
+// }
+
```
Comment (ntbm) [inline src/components/library/LibraryMangaGrid.tsx:49]:
```suggestion

```

File: src/components/MangaCard.tsx [hunk 2]
```diff
@@ -40,17 +40,21 @@ const MangaTitle = styled(Typography)({
     textShadow: '0px 0px 3px #000000',
 });
 
-const UnreadBadge = styled(Typography)(({ theme }) => ({
+const BadgeConatiner = styled('div')({
```
Comment (ntbm) [inline src/components/MangaCard.tsx:43]: Typo :)
```suggestion
const BadgeContainer = styled('div')({
```

File: src/components/MangaCard.tsx [hunk 3]
```diff
@@ -93,12 +97,24 @@ const MangaCard = React.forwardRef<HTMLDivElement, IProps>((props: IProps, ref)
                             height: '100%',
                         }}
                     >
-                        {unread! > 0
-                            && (
-                                <UnreadBadge>
+
+                        <BadgeConatiner>
```
Comment (ntbm) [inline src/components/MangaCard.tsx:101]: Typo
```suggestion
                        <BadgeContainer>
```

File: src/components/MangaCard.tsx [hunk 4]
```diff
@@ -93,12 +97,24 @@ const MangaCard = React.forwardRef<HTMLDivElement, IProps>((props: IProps, ref)
...
+                            )}
+                            {downloadCount! > 0 && (
+                                <Typography sx={{
+                                    backgroundColor: 'success.dark',
+                                }}
+                                >
+                                    {downloadCount}
+                                </Typography>
+                            )}
+                        </BadgeConatiner>
```
Comment (ntbm) [inline src/components/MangaCard.tsx:117]:
```suggestion
                        </BadgeContainer>
```

File: src/components/library/LibraryOptions.tsx [hunk 5]
```diff
@@ -13,10 +13,13 @@ import useLibraryOptions from 'util/useLibraryOptions';
 import ThreeStateCheckbox from 'components/util/ThreeStateCheckbox';
 
 function Options() {
-    const { unread, setUnread } = useLibraryOptions();
+    const {
+        downloaded, setDownloaded, unread, setUnread,
+    } = useLibraryOptions();
     return (
-        <div>
+        <div style={{ display: 'Flex', flexDirection: 'column' }}>
```
Comment (ntbm) [inline src/components/library/LibraryOptions.tsx:20]: :)
```suggestion
        <div style={{ display: 'flex', flexDirection: 'column' }}>
```

File: src/util/useLibraryOptions.ts [hunk 6]
```diff
@@ -20,19 +22,23 @@ interface IUseLibraryOptions {
...
+    };
     const setUnread = (newUnread: NullAndUndefined<boolean>) => {
         setSearchQuery(Object.assign(searchQuery, { unread: newUnread }), 'replace');
     };
     const setQuery = (newQuery: NullAndUndefined<string>) => {
         setSearchQuery(Object.assign(searchQuery, { query: newQuery }), 'replace');
     };
     // eslint-disable-next-line eqeqeq
-    const active = !(unread == undefined);
+    const active = !(unread == undefined) && !(downloaded == undefined);
```
Comment (ntbm) [inline src/util/useLibraryOptions.ts:40]:
```suggestion
    const active = !(unread == undefined) || !(downloaded == undefined);
```
It's active when there's a value for the unread filter or the dowloaded filter

**Review by ntbm — CHANGES_REQUESTED** [review]

File: src/components/library/LibraryMangaGrid.tsx [hunk 7]
```diff
@@ -23,15 +23,41 @@ function unreadFilter(unread: NullAndUndefined<boolean>, { unreadCount }: IManga
...
+//     { downloadCount }: IMangaCard): boolean {
+//     switch (downloaded) {
+//         case true:
+//             return !!downloaded && downloadCount >= 1;
+//         case false:
+//             return downloadCount === 0;
+//         default:
+//             return true;
+//     }
+// }
```
Comment (AriaMoradi) [inline src/components/library/LibraryMangaGrid.tsx]: No commented code please.

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: src/components/library/LibraryMangaGrid.tsx (hunk 7 above)
Comment (AriaMoradi) [reply src/components/library/LibraryMangaGrid.tsx]: git remembers for you...

**Comment by abhijeetChawla** [comment]
Changed the Navbar colors in Light mode to make it more legibile
Before
localhost_3000_library (2)
localhost_3000_library (3)

After
localhost_3000_library (4)
localhost_3000_library (1)

**Comment by AriaMoradi** [comment]
> Changed the Navbar colors in Light mode to make it more legibile

This commit is not related to the subject of this one, put it in another PR

**Review by AriaMoradi — APPROVED** [review]

**Pull request merged**